zerver/models.py

```diff
@@ -1089,6 +1090,16 @@ def is_claimed(self):
     def get_url(self):
         return "/user_uploads/%s" % (self.path_id)
 
+def validate_attachment_request(user_profile, path_id):
+    try:
+        attachment = Attachment.objects.get(path_id = path_id)
+        if (user_profile in attachment.recipients.all()) or user_profile == attachment.owner:
```

Comment:
This will perform poorly if an attachment has a large number of recipients, since we need to fetch them all each time.

Can you write this to instead just check if the user_profile is in the recipients query in the Django query system?